devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java

```diff
@@ -32,14 +32,18 @@ public boolean addExtensions(final Set<String> extensions) throws IOException {
         boolean updated = false;
         List<Dependency> dependenciesFromBom = getDependenciesFromBom();
 
```

Comment:
So I took a look at it yesterday evening and I think we should change the whole logic to be more consistent.

I would go with this algorithm:
- if dependency does not contain `:`
  - have the groupId be `io.quarkus`
  - if dependency does not start with `quarkus-`, add `quarkus-` to the artifactId
  - then create the dependency with `extension.toDependency(containsBOM(model) && isDefinedInBom(dependenciesFromBom, extension))`
- else
  - parse it with MojoUtils.parse(dependency);

The idea of this first part is to have a totally consistent dependency.

Then I would:
- if groupId is `io.quarkus`, check if it is a proper extension that is in our list. If not, I would throw an error ("Cannot find a Quarkus extension matching..."). (Let's allow the addition of external dependency, that's why we have the groupId check)
- check if the artifact is already present in the pom (right now, we only test if we found an extension in the first step, we should do it even if the extension contains `:`) and log the skip warning
- add the extension if everything went well

The only thing that I'm not totally sure about is about the `containsBOM/isDefinedInBom`. I think it should also be taken into account if the dependency has a `:` and is a Quarkus one: we don't want a dependency that is not consistent with the bom version, and if the version is forced on us, we should probably override it and log a warning.

Does it make sense to you? Would you be up for continuing the work along these lines?

Reply:
lgtm, so i will implement it, i hope finish this night ;o). Next available time will be this weekend.
thanks for this return